packages/flutter_tools/lib/src/ios/fallback_discovery.dart

```diff
@@ -132,8 +132,12 @@ class FallbackDiscovery {
         final VmService vmService = await _vmServiceConnectUri(assumedWsUri.toString());
         final VM vm = await vmService.getVM();
         for (final IsolateRef isolateRefs in vm.isolates) {
-          final Isolate isolate = await vmService.getIsolate(isolateRefs.id) as Isolate;
-          final LibraryRef library = isolate.rootLib;
+          final dynamic isolateResponse = await vmService.getIsolate(isolateRefs.id);
+          if (isolateResponse is! Isolate) {
+            // Might have been a Sentinel. Try again later.
+            throw Exception();
```

Comment:
Might be worth a better exception message?

Reply:
Or a check that it is a Sentinel instead?

Reply:
Made check for Sentinel and added error message to expception

Reply:
exception